homeassistant/components/integration/sensor.py

```diff
@@ -89,13 +121,14 @@ async def async_setup_platform(
 ) -> None:
     """Set up the integration sensor."""
     integral = IntegrationSensor(
-        config[CONF_SOURCE_SENSOR],
-        config.get(CONF_NAME),
-        config[CONF_ROUND_DIGITS],
-        config[CONF_UNIT_PREFIX],
-        config[CONF_UNIT_TIME],
-        config.get(CONF_UNIT_OF_MEASUREMENT),
-        config[CONF_METHOD],
+        integration_method=config[CONF_METHOD],
+        name=config.get(CONF_NAME),
+        round_digits=config[CONF_ROUND_DIGITS],
+        source_entity=config[CONF_SOURCE_SENSOR],
+        unique_id=None,
```

Comment:
Can we piggyback adding unique_id to configuration yaml :) ?

Reply:
Let's do that in a different PR.